config/eslint/eslintrc.js

```diff
@@ -0,0 +1,205 @@
+        trailingUnderscore: "allow",
+      },
+      {
+        selector: "classProperty",
+        format: ["camelCase", "UPPER_CASE"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "enumMember",
+        format: ["UPPER_CASE"],
+      },
+      {
+        selector: "memberLike",
+        modifiers: ["private"],
+        format: ["camelCase"],
+        leadingUnderscore: "require",
+      },
+      {
+        selector: ["objectLiteralProperty", "objectLiteralMethod"],
+        format: ["camelCase", "PascalCase", "snake_case", "UPPER_CASE"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "typeProperty",
+        format: ["camelCase", "PascalCase"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "typeLike",
+        format: ["PascalCase"],
+      },
+      {
+        selector: "typeProperty",
+        filter: "__hardhatContext",
+        format: null,
+      },
+    ],
+    "@typescript-eslint/no-empty-interface": "error",
+    "@typescript-eslint/no-floating-promises": "error",
+    "@typescript-eslint/no-misused-new": "error",
+    "@typescript-eslint/no-namespace": "error",
+    "@typescript-eslint/no-redeclare": "error",
+    "@typescript-eslint/no-shadow": [
+      "error",
+      {
+        hoist: "all",
+      },
+    ],
+    "@typescript-eslint/no-this-alias": "error",
+    "@typescript-eslint/no-unused-expressions": "error",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        argsIgnorePattern: "^_",
+        varsIgnorePattern: "^_",
+      },
+    ],
+    "@typescript-eslint/prefer-for-of": "error",
+    "@typescript-eslint/prefer-function-type": "error",
+    "@typescript-eslint/prefer-namespace-keyword": "error",
+    "@typescript-eslint/restrict-plus-operands": "error",
+    "@typescript-eslint/strict-boolean-expressions": [
+      "error",
+      {
+        allowAny: true,
+      },
+    ],
+    "@typescript-eslint/triple-slash-reference": [
+      "error",
+      {
+        path: "always",
+        types: "prefer-import",
+        lib: "always",
+      },
+    ],
+    "@typescript-eslint/unified-signatures": "error",
+    "constructor-super": "error",
+    eqeqeq: ["error", "always"],
+    "guard-for-in": "error",
+    "id-blacklist": "error",
+    "id-match": "error",
+    "import/no-extraneous-dependencies": ["error", {
+      devDependencies: false,
+    }],
+    "import/order": [
+      "error",
+      {
+        groups: [
+          "object",
+          ["builtin", "external"],
+          "parent",
+          "sibling",
+          "index",
+        ],
+      },
+    ],
+    "no-bitwise": "error",
+    "no-caller": "error",
+    "no-cond-assign": "error",
+    "no-debugger": "error",
+    "no-duplicate-case": "error",
+    "no-duplicate-imports": "error",
+    "no-eval": "error",
+    "no-extra-bind": "error",
+    "no-new-func": "error",
+    "no-new-wrappers": "error",
+    "no-return-await": "error",
```

Comment:
I'm ok with this rule for now, but TBH I'd rather enforce awaits everywhere, as that produces better async stack traces.

https://twitter.com/a0viedo/status/1305911417355333633

Reply:
Makes sense to me: https://github.com/nomiclabs/hardhat/issues/1596